prov/sockets/src/sock_ep.c

```diff
@@ -584,7 +584,10 @@ static ssize_t sock_tx_size_left(struct fid_ep *ep)
 	fastlock_acquire(&tx_ctx->wlock);
 	num_left = rbavail(&tx_ctx->rb)/SOCK_EP_TX_ENTRY_SZ;
 	fastlock_release(&tx_ctx->wlock);
-	return num_left;
+	if (num_left > SOCK_EP_TX_SZ)
```

Comment:
> Probably returning 368 is alright since socket provider already have the buffer space to accommodate 368. What do you think?

Yeah, I think it's fine. I'll modify the fabtest PR to make sure that it is greater than or equal to the size provided in hints.